Neos.Flow/Classes/Mvc/Routing/Router.php

```diff
@@ -267,9 +278,86 @@ protected function createRoutesFromConfiguration()
             }
             $this->routes[] = $route;
         }
+
         $this->routesCreated = true;
     }
 
+    public function initializeAnnotatedRoutes(): array
+    {
+        return static::resolveRouteAnnotatedMethods($this->objectManager);
+    }
+
+
+    /**
+     * @param ObjectManagerInterface $objectManager
+     * @Flow\CompileStatic
+     * @return array
+     * @throws InvalidAnnotatedMethodException if a none-Action method is annotated
+     */
+    protected static function resolveRouteAnnotatedMethods(ObjectManagerInterface $objectManager): array
+    {
+        $annotatedRoutes = [];
+        $reflectionService = $objectManager->get(ReflectionService::class);
+        $controllerClassNames = $reflectionService->getAllSubClassNamesForClass(AbstractController::class);
+        foreach ($controllerClassNames as $controllerClassName) {
+            if ($reflectionService->isClassAbstract($controllerClassName)) {
+                continue;
+            }
+
+            $methodNames = $reflectionService->getMethodsAnnotatedWith($controllerClassName, Flow\Route::class);
+            foreach ($methodNames as $methodName) {
+                if (preg_match('/.*Action$/', $methodName) === 0 || $reflectionService->isMethodPublic($controllerClassName, $methodName) === false) {
+                    throw new InvalidAnnotatedMethodException(sprintf('The method %s->%s is annotated with @Flow\Route. The Route annotation is only meant for *Action methods in your controller', $controllerClassName, $methodName));
+                }
+                $controllerObjectName = $objectManager->getCaseSensitiveObjectName($controllerClassName);
+
+                if ($controllerObjectName === null) {
+                    throw new UnknownObjectException(sprintf('The object "%s" is not registered.', $controllerClassName), 1618384920);
+                }
+                $controllerPackageKey = $objectManager->getPackageKeyByObjectName($controllerObjectName);
+
+                $matches = [];
+                $subject = substr($controllerObjectName, strlen($controllerPackageKey) + 1);
```

Comment:
While trying to extract this to a factory (as discussed via Slack) I wonder: What is this line supposed to do?
It shortens the controller class name by the length of the package key, but that doesn't have to reflect the namespace

Reply:
..and now I realized that this, too, is taken 1:1 from the `ActionRequest` 😬
Even more a reason to centralize (and fix) this piece of logic.. I'll give it a try